metadata_mapper/mappers/ucsd_blacklight/ucsd_blacklight_mapper.py

```diff
@@ -0,0 +1,398 @@
+                         "Moderator", "Monitor", "Music copyist",
+                         "Musical director", "Musician", "Narrator",
+                         "Onscreen presenter", "Opponent", "Organizer",
+                         "Originator", "Other", "Owner", "Panelist",
+                         "Papermaker", "Patent applicant", "Patent holder",
+                         "Patron", "Performer", "Permitting agency",
+                         "Plaintiff", "Plaintiff-appellant",
+                         "Plaintiff-appellee", "Platemaker", "Praeses",
+                         "Presenter", "Printer", "Printer of plates",
+                         "Printmaker", "Process contact", "Producer",
+                         "Production company", "Production designer",
+                         "Production manager", "Production personnel",
+                         "Production place", "Programmer", "Project director",
+                         "Proofreader", "Provider", "Publication place",
+                         "Publishing director", "Puppeteer", "Radio director",
+                         "Radio producer", "Recording engineer", "Recordist",
+                         "Redaktor", "Renderer", "Reporter", "Repository",
+                         "Research team head", "Research team member",
+                         "Researcher", "Respondent", "Responsible party",
+                         "Restager", "Restorationist", "Reviewer", "Rubricator",
+                         "Scenarist", "Scientific advisor", "Screenwriter",
+                         "Scribe", "Sculptor", "Second party", "Secretary",
+                         "Seller", "Set designer", "Setting", "Signer",
+                         "Singer", "Sound designer", "Speaker", "Sponsor",
+                         "Stage director", "Stage manager", "Standards body",
+                         "Stereotyper", "Storyteller", "Supporting host",
+                         "Surveyor", "Teacher", "Technical director",
+                         "Television director", "Television producer",
+                         "Thesis advisor", "Transcriber", "Translator",
+                         "Type designer", "Typographer", "University place",
+                         "Vessel", "Videographer", "Voice actor", "Witness",
+                         "Wood engraver", "Woodcutter",
+                         "Writer of accompanying material",
+                         "Writer of added commentary", "Writer of added lyrics",
+                         "Writer of added text", "Writer of introduction",
+                         "Writer of preface",
+                         "Writer of supplementary textual content"]
+
+
+class UcsdBlacklightValidator(Validator):
+    def setup(self, **options):
+        self.add_validatable_field(
+            field="identifier", type=Validator.list_of(str),
+            validations=[
+                UcsdBlacklightValidator.identifier_content_match
```

Comment:
The problem with leaving the type_match validator is that the `UcsdBlacklightValidator.identifier_content_match` should prevent further validations from running, right? Maybe that's a bad assumption on my part -- let me know.

I'm calling the `Validator.content_match` only if the `UcsdBlacklightValidator` doesn't return `None`. I could call both `content_match` and `type_match` from within the `UcsdBlacklightValidator.identifier_content_match` if the presence of an expected None with an ark value doesn't dismiss them both those validations.

Reply:
@lthurston I am confused by why validators don't cascade (this is a good question for @timfrazee), but it is true that validators are atomic and don't cascade. It is also true that identifier should be a list of strings. 🤷‍♀️ Since that's the way the validator is designed, I've tried to conform to it. 

I can see how implementing cascading validators could add significant complexity, but intuitively, I think about validation as a series of cascading checks - a failure at one stage should prevent further analysis, as you suggest. It would also be convenient to rely on the assertions made in prior validations, for example, implementing a content comparison becomes much easier if we already know the type is the same.

Reply:
The short version is - in previous discussions, we had decided that exposing all validation errors, even those that are redundant, was preferred in order to prevent having to run the validator multiple times to get at all the errors. If we were to short-circuit on a failure, a later failure might not be raised.

Adding short-circuiting would be easy, but trying to do so while not obscuring real failures will be more challenging. Let's discuss in our check-in meeting.

Reply:
Ahh, I guess I did not properly understand that decision or the full implications of it when it was made. That does make some sense, though in practice we wind up running these validation reports over and over again as we iterate on both the mapping logic and custom validations, so "hidden" errors would, I believe, eventually surface. I don't think it's worth it to re-address this now - I've gotten used to thinking about it in this way, and the unit testing framework you're working on @timfrazee seems far more valuable. It does seem accurate to add the `type_match` validation back in though, no?

Reply:
There is something delightfully straightforward about validations that simply assert one thing and don't depend on each other.